website/docs/worker-events.md

```diff
@@ -0,0 +1,165 @@
+---
+
+We support a large number of events via an EventEmitter. You can either retrieve
+the event emitter via the `events` property on the `Runner` object, or you can
+create your own event emitter and pass it to Graphile Worker via the
+`WorkerOptions.events` option (this is primarily useful for receiving events
+during Graphile Worker's startup procedure, before the `run()` promise
+resolves).
+
+### Example: via `runner.events`
+
+```js
+runner.events.on("job:success", ({ worker, job }) => {
+  console.log(`Hooray! Worker ${worker.workerId} completed job ${job.id}`);
+});
+```
+
+### Example: using `EventEmitter`
+
+```js
+const events = new EventEmitter();
+events.on(...);
+events.on(...);
+events.on(...);
+
+const runner = await run({ events, ... });
+```
+
+### Example: In graphile configuration
+
+```js
+const events = new EventEmitter();
+events.on(...);
+events.on(...);
+events.on(...);
+
+const preset = {
+  worker: {
+    events,
+    ...
+  }
+};
+```
+
+## Definitions:
+
+Details of what events we support and what data is available on the event
+payload is detailed below in TypeScript syntax:
+
+```ts
+export type WorkerEvents = TypedEventEmitter<{
+  /**
+   * When a worker pool is created
+   */
+  "pool:create": { workerPool: WorkerPool };
+
+  /**
+   * When a worker pool attempts to connect to PG ready to issue a LISTEN
+   * statement
+   */
+  "pool:listen:connecting": { workerPool: WorkerPool };
+
+  /**
+   * When a worker pool starts listening for jobs via PG LISTEN
+   */
+  "pool:listen:success": { workerPool: WorkerPool; client: PoolClient };
+
+  /**
+   * When a worker pool faces an error on their PG LISTEN client
+   */
+  "pool:listen:error": {
+    workerPool: WorkerPool;
+    error: any;
+    client: PoolClient;
+  };
+
+  /**
+   * When a worker pool is released
+   */
+  "pool:release": { pool: WorkerPool };
+
+  /**
+   * When a worker pool starts a graceful shutdown
+   */
+  "pool:gracefulShutdown": { pool: WorkerPool; message: string };
+
+  /**
+   * When a worker pool graceful shutdown throws an error
+   */
+  "pool:gracefulShutdown:error": { pool: WorkerPool; error: any };
+
+  /**
+   * When a worker is created
+   */
+  "worker:create": { worker: Worker; tasks: TaskList };
+
+  /**
+   * When a worker release is requested
+   */
+  "worker:release": { worker: Worker };
+
+  /**
+   * When a worker stops (normally after a release)
+   */
+  "worker:stop": { worker: Worker; error?: any };
+
+  /**
+   * When a worker is about to ask the database for a job to execute
+   */
+  "worker:getJob:start": { worker: Worker };
+
+  /**
+   * When a worker calls get_job but there are no available jobs
+   */
+  "worker:getJob:error": { worker: Worker; error: any };
+
+  /**
+   * When a worker calls get_job but there are no available jobs
+   */
+  "worker:getJob:empty": { worker: Worker };
+
+  /**
+   * When a worker is created
+   */
+  "worker:fatalError": { worker: Worker; error: any; jobError: any | null };
```

Comment:
Also I've noticed that this comment is probably wrong. Do you have a suggestion for a better documentation?

I had a look at the code, is it when the worker cannot be created? In that case, what would you think of this proposition? 

```suggestion
  /**
   * When a worker cannot be created because of an unhandled error
   */
  "worker:fatalError": { worker: Worker; error: any; jobError: any | null };
```